### PR TITLE
Make tests leaner, concrete and slightly more understandable

### PR DESCRIPTION
**Pull request by ragunathjawahar**

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt [hunk 1]
```diff
@@ -101,15 +96,6 @@ class PatientEditScreenControllerTest {
     val address = PatientMocker.address()
     val phoneNumber: PatientPhoneNumber? = null
 
-    whenever(patientRepository.patient(patient.uuid)).thenReturn(Observable.just(patient.toOptional()))
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt]: 👏 👏 👏

File: app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt [hunk 2]
```diff
@@ -101,15 +96,6 @@ class PatientEditScreenControllerTest {
...
-    whenever(patientRepository.address(address.uuid)).thenReturn(Observable.just(address.toOptional()))
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt]: We can remove these stubs from `PatientEditScreenValidationUsingMockDateValidator` and `PatientEditScreenValidationUsingMockNumberValidatorTest` as well.

Comment (ragunathjawahar) [reply app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt]: That is coming in as part of the next PR.

File: app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt [hunk 3]
```diff
@@ -312,456 +312,488 @@ class PatientEditScreenControllerTest {
...
-                addressUuid = UUID.fromString("6953267f-23d0-4ec7-9623-0dbd6c264189"),
-                dateOfBirthString = "1995-06-13"
-            ),
-            inputEvents = listOf(
-                PatientEditDateOfBirthTextChanged(""),
-                PatientEditAgeTextChanged("30"),
-                PatientEditAgeTextChanged(""),
-                PatientEditDateOfBirthTextChanged("13/06/1995")),
-            shouldShowConfirmDiscardChangesPopup = false))
+        noUserInputOnScreen,
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/editpatient/PatientEditScreenControllerTest.kt]: Nice!

**Pull request merged**